src/template.ts

```diff
@@ -165,6 +165,27 @@ template.innerHTML = `
       transform: translateY(-100%);
     }
 
+    @keyframes wiggle {
+      10%, 12% {
+        transform: translateX(-5%);
```

Comment:
Building off the 3rd option, and offering a CSS custom property to override the `animation-name` property. Instead of `animation-name`, we can simply add the override for `animation` which is the shorthand to all the animation properties. This way the user can specify their own animation keyframes (name), animation duration, delay, iteration count, etc... by overriding this exposed property.

CSS Syntax:
```css
animation: name duration timing-function delay iteration-count direction fill-mode play-state;
```

Reply:
@maaslalani it's a nice idea, but unfortunately shadow root encapsulation will not allow for this. If I specify a keyframe animation in the global document scope, it won't be available in the local shadow scope. The user would be able to change the name that is sued, but CSS won't be able to find the keyframes that correspond to that name (except in legacy polyfill scenarios, where encapsulation is just a facade). [Here is an example that demonstrates the problem](http://lopsided-motion.glitch.me/animation-encapsulation.html).

Reply:
I suppose there is precedence for the attribute approach. We have the whole `allow-when-focused` and `always-allow` which are opinionated ways of interacting with the viewer. 

So it comes with `basic` and `wiggle` out of the box, but it would be nice to be able to add your own.

Reply:
> So it comes with basic and wiggle out of the box, but it would be nice to be able to add your own.

@pushmatrix agreed, and this is why this change is relatively difficult to land as-is. The UX is quite opinionated and difficult to customize. Ideally we would lean in to something like #738 to accomplish specialty interaction prompts like this.

As mentioned above, CSS parts would make this effect a lot easier to customize. For now, we'll have to wait for future web platform features to land, and try to get to #738 as quickly as we can.

Reply:
The WebKit folks have blessed us with some promising news on this topic: https://webkit.org/blog/9609/release-notes-for-safari-technology-preview-94/

> Added support for the ::part() pseudo element from CSS Shadow Parts